packages/material-ui/src/IconButton/IconButton.test.js

```diff
@@ -107,4 +108,22 @@ describe('<IconButton />', () => {
       assert.strictEqual(ReactDOM.findDOMNode(ref.args[0][0]).type, 'button');
     });
   });
+
+  describe('Firefox onClick', () => {
+    beforeEach(() => {
+      consoleErrorMock.spy();
+    });
+
+    afterEach(() => {
+      consoleErrorMock.reset();
+    });
+
+    it('should raise a warning', () => {
+      shallow(
```

Comment:
I think `shallow` does not trigger propTypes warnings. You would need to mount and then inspect that consoleErrorMock logged to correct warning.

Reply:
It does. We wouldn't have 100% test coverage otherwise.